### PR TITLE
#2.11 implementera chunks

### PR DESCRIPTION
**Pull request by Manfred-Hastmark**
Chunks are now implemented.

image

### CODE REVIEW

File: Assets/Scripts/MarchingCubes/MarchingCubes.cs [hunk 1]
```diff
@@ -10,42 +11,37 @@ public class MarchingCubes
...
     readonly float amplitude;
-    readonly float bottomLevel;
-
-    Mesh mesh;
+    public int chunkResolution;
 
     /// <summary>
     /// Initializes the MarchingCubes script
     /// </summary>
     /// <param name="mesh"></param>
```
Comment (isakswe) [inline Assets/Scripts/MarchingCubes/MarchingCubes.cs:22]: Seems to be some old documentation. Either document the parameters properly or just remove them and just keep the method summary.

File: Assets/Scripts/MarchingCubes/Chunk.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    PillPlayerController player;
+    MinMaxTerrainLevel terrainLevel;
+
+    /// <summary>
+    /// Initialize
+    /// </summary>
+    /// <param name="index"></param>
+    /// <param name="resolution"></param>
+    /// <param name="marchingCubes"></param>
+    /// <param name="player"></param>
```
Comment (isakswe) [inline Assets/Scripts/MarchingCubes/Chunk.cs:24]: Empty and seemingly out od date param summary. Either leave out or document

File: Assets/Scripts/MarchingCubes/Chunk.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public class Chunk : MonoBehaviour
+{
+
+    int index;
+    int resolution;
+    MeshFilter meshFilter;
+    MeshCollider meshCollider;
+    Mesh mesh;
+    MarchingCubes marchingCubes;
+    PillPlayerController player;
```
Comment (isakswe) [inline Assets/Scripts/MarchingCubes/Chunk.cs:15]: Not used but creates a potential future dependency on PillPlayerController.

File: Assets/Scripts/MarchingCubes/Planet.cs [hunk 4]
```diff
@@ -9,25 +9,35 @@ public class Planet : MonoBehaviour
...
 
+    public float diameter;
     public float radius;
     public float surfaceGravity;
     public string bodyName = "TBT";
     public float mass;
     public List<Planet> moons;
 
+    private List<Chunk> chunks;
+    private PillPlayerController player;
```
Comment (isakswe) [inline Assets/Scripts/MarchingCubes/Planet.cs:26]: Unused and potential future dependency on PillPlayerController

File: Assets/Scripts/MarchingCubes/SpawnPlanet.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+        planet.transform.position = new Vector3(100, 0, 0);
+
+        Universe.InitializeRandomWithSeed();
+        planet.Initialize(player, 1);
+        planet.SetUpPlanetValues();
+
+        player.Initialize(planet.gameObject);
+    }
+}
```
Comment (isakswe) [inline Assets/Scripts/MarchingCubes/SpawnPlanet.cs:26]: Don't know what this file is for. Testing?

**Review by isakswe — CHANGES_REQUESTED** [review]
Some stuff about comments, unused variables, and stuff

File: Assets/Scripts/MarchingCubes/Chunk.cs (hunk 3 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/Chunk.cs:15]: Will be used in Chunks later on to do "LOD's"

File: Assets/Scripts/MarchingCubes/Planet.cs (hunk 4 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/Planet.cs:26]: Will be used in Chunks later on to do "LOD's"

File: Assets/Scripts/MarchingCubes/SpawnPlanet.cs (hunk 5 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/SpawnPlanet.cs:26]: Indeed

File: Assets/Scripts/MarchingCubes/Chunk.cs (hunk 3 above)
Comment (isakswe) [reply Assets/Scripts/MarchingCubes/Chunk.cs:15]: Change it to GameObject or even Transform of player to keep it abstract?

Comment (isakswe) [reply Assets/Scripts/MarchingCubes/Chunk.cs:15]: Goes for all similar comments

File: Assets/Scripts/MarchingCubes/SpawnPlanet.cs (hunk 5 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/SpawnPlanet.cs:26]: It's Nice to have to be able to get a planet in your own scene, I don't see a problem keeping it

File: Assets/Scripts/MarchingCubes/MarchingCubes.cs (hunk 1 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/MarchingCubes.cs]: fixed

File: Assets/Scripts/MarchingCubes/Chunk.cs (hunk 2 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/Chunk.cs]: fixed

File: Assets/Scripts/MarchingCubes/Chunk.cs (hunk 3 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/Chunk.cs]: fixed

File: Assets/Scripts/MarchingCubes/Planet.cs (hunk 4 above)
Comment (Manfred-Hastmark) [reply Assets/Scripts/MarchingCubes/Planet.cs]: fixed

**Review by isakswe — APPROVED** [review]

**Pull request merged**